### PR TITLE
Upgrading to Elasticsearch 6.0 and Adding sniffing connection pool

### PR DESCRIPTION
**Pull request by jeremysmith1**
The goal was to upgrade the client to 6.0. The comments in ElasticSearchOutput.cs suggested adding connection pools was desired.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -612,7 +612,7 @@ This output writes data to the [Elasticsearch](https://www.elastic.co/products/e
 | :---- | :-------------- | :------: | :---------- |
 | `type` | "ElasticSearch" | Yes | Specifies the output type. For this output, it must be "ElasticSearch". |
 | `indexNamePrefix` | string | No | Specifies the prefix to be used when creating the Elasticsearch index. This prefix, together with the date of when the data was generated, will be used to form the name of the Elasticsearch index. If not specified, a prefix will not be used. |
-| `serviceUri` | URL:port | Yes | Specifies where the Elasticsearch cluster is. This is needed for EventFlow to locate the cluster and send the data. |
+| `serviceUri` | URL:port | Yes | Specifies where the Elasticsearch cluster is. This is needed for EventFlow to locate the cluster and send the data. Single URL accepted or Semicolon seperated URLs for connection pool |
```
Comment (karolz-ms) [inline README.md]: Single URL or semicolon-separated URLs for connection pooling are accepted

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch.csproj [hunk 2]
```diff
@@ -26,8 +26,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Elasticsearch.Net" Version="5.6.0" />
-    <PackageReference Include="NEST" Version="5.6.0" />
+    <PackageReference Include="Elasticsearch.Net" Version="6.0.2" />
```
Comment (karolz-ms) [inline src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch.csproj]: Are NEST/Elasticsearch.Net 6.x backward-compatible with Elasticsearch 5 (server)?

If yes, please rev up the Outputs.Elasticsearch package version (`<VersionPrefix>` tag in the ElasticSearch.csproj file) to 2.4.0
If not, please rev up the package version to 3.0.0

**Comment by karolz-ms** [comment]
Thanks for your contribution @jeremysmith1 , much appreciated!!

File: src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch.csproj (hunk 2 above)
Comment (jeremysmith1) [reply src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch.csproj]: I did not want to push people to upgrade their elastic cluster. So I tested SendEventsAsync() data using Elasticsearch.NET 6.x client with a Elasticsearch cluster 5.6.9 in docker. Worked like a charm. =)

Updating to 2.4.0.

**Comment by karolz-ms** [comment]
Fantastic

**Comment by karolz-ms** [comment]
@jeremysmith1  this should be downloadable from Nuget shortly. Thanks again!

**Comment by jeremysmith1** [comment]
Thanks for the merge. Glad I could contribute.

**Pull request merged**